File: Cpp/1702_maximum_binary_string_after_change.cpp
// https://leetcode.cn/problems/maximum-binary-string-after-change/

class Solution {
public:
    string maximumBinaryString(string binary) {
        string res(binary.size(), '1');
        int idx = -1;
        for (int i = 0; i < binary.size(); ++i) {
            char ch = binary[i];
            if (ch == '0') {
                if (idx < 0) idx = i;
                else idx += 1;
            }
        }
        if (idx >= 0) {
            res[idx] = '0';
        }
        return res;
    }
};

File: Python3/1702_maximum_binary_string_after_change.py
# https://leetcode.cn/problems/maximum-binary-string-after-change/

class Solution:
    def maximumBinaryString(self, binary: str) -> str:
        res = ["1" for _ in range(len(binary))]
        idx = -1
        for i, ch in enumerate(binary):
            if ch == "0":
                if idx < 0:
                    idx = i
                else:
                    idx += 1
        if idx >= 0:
            res[idx] = "0"
        return "".join(res)
